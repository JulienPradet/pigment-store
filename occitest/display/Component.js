import React from 'react'
import Feature from './Feature'

export default class Component extends React.Component {
  render() {
    return <div>
      {this.props.name}

      <ul>
        {
          Object.keys(this.props.component.features)
            .map(featureKey => <li key={featureKey}>
              <Feature
                name={featureKey}
                {...this.props.component.features[featureKey]}
              />
            </li>)
        }
      </ul>
    </div>
  }
}
