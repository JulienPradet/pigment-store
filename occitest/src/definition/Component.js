import Feature from './Feature'

export default function (Component, displayName) {
  return {
    Component,
    displayName,
    features: {},
    feature(featureName, props, actions) {
      this.features[featureName] = Feature(Component, props, actions)
      return this
    },
    document(props) {
      this.features.__doc = Feature(Component, props)
      return this
    }
  }
}
