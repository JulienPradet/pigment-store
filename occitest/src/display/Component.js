import React from 'react'
import Feature from './Feature'
import {colors} from './styles'

const styles = {
  container: {
    margin: '0',
  },
  title: {
    margin: 0,
    fontWeight: 'bold',
    fontSize: '1.2em'
  },
  subTitle: {
    margin: 0,
    fontWeight: 'bold',
    fontSize: '1.1em'
  },
  strong: {
    margin: 0,
    fontWeight: 'bold',
    fontSize: '1em'
  },
  featureList: {
    padding: 0,
    margin: '1em 0',
    border: '1px solid ' + colors.white,
    background: colors.white
  },
  featureContainer: {
    display: 'block'
  }
}

export default class Component extends React.Component {
  constructor() {
    super()
    this.state = {}
    this.showComponentDetails = this.showComponentDetails.bind(this)
    this.hideComponentDetails = this.hideComponentDetails.bind(this)
    this.toggleComponentDetails = this.toggleComponentDetails.bind(this)
  }

  showComponentDetails() {
    this.setState({showDetails: true})
  }

  hideComponentDetails() {
    this.setState({showDetails: false})
  }

  toggleComponentDetails() {
    if(this.state.showDetails) {
      this.hideComponentDetails()
    } else {
      this.showComponentDetails()
    }
  }

  render() {
    return <div style={styles.container}>
      <h3 style={styles.title}>{this.props.name}</h3>
      <button onClick={this.toggleComponentDetails}>Show Details</button>

      <ul style={styles.featureList}>
        {Object.keys(this.props.component.features)
            .filter((featureKey) => featureKey !== '__doc')
            .map(featureKey => <li key={featureKey} style={styles.featureContainer}>
              <Feature
                name={featureKey}
                detailed={this.state.showDetails}
                {...this.props.component.features[featureKey]}
              />
            </li>)}
      </ul>
    </div>
  }
}
