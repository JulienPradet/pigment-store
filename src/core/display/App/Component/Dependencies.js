import React from 'react'
import {Link} from 'react-router'
import * as StackedList from '../util/View/StackedList'
import * as LinkList from '../util/View/LinkList'

const makePath = (prefix, type, path) => {
  return `${prefix}/${type}-${path}`
}

const stackDependencies = (dependencies = []) => {
  const firstLevelStackObject = dependencies
    .filter((path) => path.length > 0)
    .map((path) => ({
      path: path[0],
      children: path.slice(1).filter((path) => path.length > 0)
    }))
    .reduce((stacks, {path, children}) => {
      const pathStack = children.length > 0
        ? stacks.hasOwnProperty(path)
          ? [...stacks[path], children]
          : [children]
        : []

      return Object.assign(
        {},
        stacks,
        {[path]: pathStack}
      )
    }, {})

  const stackedDependencies = Object
    .keys(firstLevelStackObject)
    .map((key) => ({
      path: key,
      type: firstLevelStackObject[key].length > 0 ? 'category' : 'component',
      children: firstLevelStackObject[key]
    }))
    .map(({path, type, children}) => ({
      path,
      type,
      children: stackDependencies(children)
    }))

  return stackedDependencies
}

const StackedDependencyList = ({prefix = '/docs', stackedDependencies}) => <StackedList.Container>
  {stackedDependencies
    .map(({path, type, children}, index) => <StackedList.Row key={index}>
      <StackedList.Item>
        <Link to={makePath(prefix, type, path)}>{path}</Link>
      </StackedList.Item>
      {children && children.length > 0
        ? <StackedList.Item>
          <StackedDependencyList prefix={makePath(prefix, type, path)} stackedDependencies={children} />
        </StackedList.Item>
        : null}
    </StackedList.Row>)}
</StackedList.Container>

const DependencyList = ({dependencies}) => <StackedDependencyList
  stackedDependencies={stackDependencies(dependencies.map(({path}) => path))}
/>

export default ({component}) => <div>
  {Object.keys(component.Component.__PIGMENT_META.reliesOn || {}).length > 0
    ? <LinkList.Container>
      <LinkList.Title>Relies on</LinkList.Title>
      <LinkList.Content>
        <DependencyList dependencies={component.Component.__PIGMENT_META.reliesOn} />
      </LinkList.Content>
    </LinkList.Container>
    : null}
  {Object.keys(component.Component.__PIGMENT_META.isReliedOnBy || {}).length > 0
    ? <LinkList.Container>
      <LinkList.Title>Is relied on by</LinkList.Title>
      <LinkList.Content>
        <DependencyList dependencies={component.Component.__PIGMENT_META.isReliedOnBy} />
      </LinkList.Content>
    </LinkList.Container>
    : null}
</div>
