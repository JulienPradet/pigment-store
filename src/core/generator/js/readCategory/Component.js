module.exports = class Component {
  constructor (file) {
    this.file = file
  }

  render () {
    return `require('${this.file}').default`
  }
}
