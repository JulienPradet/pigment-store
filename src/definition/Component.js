import Feature from './Feature'

export default function (Component) {
  return {
    features: {},
    feature(featureName, props, state) {
      this.features[featureName] = Feature(Component, props, state)
      return this
    }
  }
}
