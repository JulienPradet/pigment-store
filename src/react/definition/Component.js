import Feature from './Feature'

export default (name, Component) => {
  return {
    Component,
    name,
    description: null,
    setDescription (description) {
      this.description = description
      return this
    },
    decorators: [],
    addDecorator (decorator) {
      this.decorators.push(decorator)
      return this
    },
    features: [],
    feature (featureName, props, actions) {
      this.features.push(Feature(Component, featureName, props, actions))
      return this
    },
    featureJsx (featureName, render, actions) {
      let jsx
      if (typeof render === 'function') {
        jsx = render()
      } else {
        console.warn('Your feature should be defered in a function. Unexpected behaviors might occurs if not.')
        jsx = render
      }
      this.features.push(Feature(jsx.type, featureName, jsx.props, actions))
      return this
    }
  }
}
