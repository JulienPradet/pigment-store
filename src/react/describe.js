import describe from './definition/Component'

export default describe
