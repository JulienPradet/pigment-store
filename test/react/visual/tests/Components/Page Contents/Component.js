import React from 'react'
import describe from 'pigment-store/react/describe'
import Component from '../../../../../../src/core/display/App/Component/Current'
import rendererDecorator from '../../__decorators__/.renderer.decorator'
import routerDecorator from '../../__decorators__/.router.decorator'
import configDecorator from '../../__decorators__/.config.decorator'
import displayOptionsDecorator from '../../__decorators__/.displayOptions.decorator'
import fixtures from '../__fixtures__/index'

export default describe('Component', Component)
  .addDecorator(rendererDecorator)
  .addDecorator(routerDecorator)
  .addDecorator(configDecorator)
  .addDecorator(displayOptionsDecorator)
  .featureJsx(
    'Default',
    () => (
      <Component pathname='' component={fixtures.categories[0].components[0]} />
    )
  )
  .featureJsx(
    'With multiple features',
    () => (
      <Component pathname='' component={fixtures.categories[0].components[1]} />
    )
  )
