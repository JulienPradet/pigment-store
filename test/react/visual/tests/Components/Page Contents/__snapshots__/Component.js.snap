exports[`Component Default 1`] = `
<div>
  <div
    className="container"
    id="component">
    <div
      className="titleContainer"
      onClick={[Function]}>
      <div
        className="title">
        <div
          className="titleContent">
          <div>
            <h1>
              Title
            </h1>
            <div
              className="meta">
              test/react/visual/tests/Components/__fixtures__/Title
            </div>
          </div>
        </div>
        <div
          className="titleButton">
          <button
            onClick={[Function]}>
            <i
              className="material-icons">
              keyboard_arrow_down
            </i>
          </button>
        </div>
      </div>
    </div>
    <div />
  </div>
  <div>
    <div
      className="container"
      id={undefined}>
      <div
        className="titleContainer"
        onClick={[Function]}>
        <div
          className="title">
          <div
            className="titleContent">
            <h2>
              <a
                className=""
                href="/full/feature-Default"
                onClick={[Function]}
                style={Object {}}>
                Default
              </a>
            </h2>
          </div>
          <div
            className="titleButton">
            <button
              onClick={[Function]}>
              <i
                className="material-icons">
                keyboard_arrow_down
              </i>
            </button>
          </div>
        </div>
        <div
          className="content">
          <div
            style={
              Object {
                "height": "100%",
                "margin": "0 auto",
                "width": "100%",
              }
            }>
            <iframe
              src="iframe.html#/feature-Default"
              style={
                Object {
                  "border": 0,
                  "height": "100px",
                  "transform": "scale(1)",
                  "transformOrigin": "top left",
                  "width": "100%",
                }
              } />
          </div>
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`Component With multiple features 1`] = `
<div>
  <div
    className="container"
    id="component">
    <div
      className="titleContainer"
      onClick={[Function]}>
      <div
        className="title">
        <div
          className="titleContent">
          <div>
            <h1>
              Avatar
            </h1>
            <div
              className="meta">
              test/react/visual/tests/Components/__fixtures__/Avatar
            </div>
          </div>
        </div>
        <div
          className="titleButton">
          <button
            onClick={[Function]}>
            <i
              className="material-icons">
              keyboard_arrow_down
            </i>
          </button>
        </div>
      </div>
    </div>
    <div />
  </div>
  <div>
    <div
      className="container"
      id={undefined}>
      <div
        className="titleContainer"
        onClick={[Function]}>
        <div
          className="title">
          <div
            className="titleContent">
            <h2>
              <a
                className=""
                href="/full/feature-Default"
                onClick={[Function]}
                style={Object {}}>
                Default
              </a>
            </h2>
          </div>
          <div
            className="titleButton">
            <button
              onClick={[Function]}>
              <i
                className="material-icons">
                keyboard_arrow_down
              </i>
            </button>
          </div>
        </div>
        <div
          className="content">
          <div
            style={
              Object {
                "height": "100%",
                "margin": "0 auto",
                "width": "100%",
              }
            }>
            <iframe
              src="iframe.html#/feature-Default"
              style={
                Object {
                  "border": 0,
                  "height": "100px",
                  "transform": "scale(1)",
                  "transformOrigin": "top left",
                  "width": "100%",
                }
              } />
          </div>
        </div>
      </div>
      <div />
    </div>
    <div
      className="container"
      id={undefined}>
      <div
        className="titleContainer"
        onClick={[Function]}>
        <div
          className="title">
          <div
            className="titleContent">
            <h2>
              <a
                className=""
                href="/full/feature-Placeholder"
                onClick={[Function]}
                style={Object {}}>
                Placeholder
              </a>
            </h2>
          </div>
          <div
            className="titleButton">
            <button
              onClick={[Function]}>
              <i
                className="material-icons">
                keyboard_arrow_down
              </i>
            </button>
          </div>
        </div>
        <div
          className="content">
          <div
            style={
              Object {
                "height": "100%",
                "margin": "0 auto",
                "width": "100%",
              }
            }>
            <iframe
              src="iframe.html#/feature-Placeholder"
              style={
                Object {
                  "border": 0,
                  "height": "100px",
                  "transform": "scale(1)",
                  "transformOrigin": "top left",
                  "width": "100%",
                }
              } />
          </div>
        </div>
      </div>
      <div />
    </div>
  </div>
</div>
`;
